frontend/packages/multicluster-sdk/src/api/useFleetK8sWatchResource.ts

```diff
@@ -101,25 +97,60 @@ export const useFleetK8sWatchResource: UseFleetK8sWatchResource = <
     ]
   )
 
-  const [data, setData] = useState<R>(fleetResourceCache[requestPath] ?? noCachedValue)
-  const [loaded, setLoaded] = useState<boolean>(!!fleetResourceCache[requestPath])
-  const [error, setError] = useState<any>(undefined)
+  // Subscribe to store changes for this specific resource path
+  const cachedEntry = useFleetK8sWatchResourceStore((state) => state.resourceCache[requestPath])
+
+  // Get store instance for imperative operations
+  const store = useFleetK8sWatchResourceStore.getState()
+
+  const [data, setData] = useState<R>(cachedEntry?.data ?? noCachedValue)
+  const [loaded, setLoaded] = useState<boolean>(cachedEntry?.loaded ?? false)
+  const [error, setError] = useState<any>(cachedEntry?.error)
+
+  // Update local state when store changes
+  useEffect(() => {
+    if (cachedEntry) {
+      setData(cachedEntry.data ?? noCachedValue)
+      setLoaded(cachedEntry.loaded)
+      setError(cachedEntry.error)
+    } else {
+      setData(noCachedValue)
+      setLoaded(false)
+      setError(undefined)
+    }
+  }, [cachedEntry, noCachedValue])
 
   useEffect(() => {
     let socket: WebSocket | undefined
     const fetchData = async () => {
-      setError(undefined)
       if (!useFleet || nullResource) {
-        setData(noCachedValue)
-        setLoaded(false)
+        store.setResource(requestPath, noCachedValue, false)
         return
       }
 
-      if (!backendPathLoaded || fleetResourceCache[requestPath]) {
+      if (!backendPathLoaded) {
+        return
+      }
+
+      // Check if we already have this resource cached and it's not expired
+      const existingEntry = store.getResource(requestPath)
+      if (existingEntry && !store.isResourceExpired(requestPath)) {
+        return
+      }
+
+      const socketKey = getCacheKey({ model, cluster, namespace, name })
+      const existingSocket = store.getSocket(socketKey)
+
+      // If we have a valid socket, increment reference count and use it
+      if (existingSocket && existingSocket.socket.readyState === WebSocket.OPEN) {
+        store.addSocketRef(socketKey)
+        socket = existingSocket.socket
```

Comment:
This is the engine behind socket re-use. A socket key is indexed using these options. When an existing key (which indicates an active socket) is found, it is reused for that hook.